app/addons/config/components.react.jsx

```diff
@@ -0,0 +1,410 @@
+        <td>{this.props.option.optionName}</td>
+        <ConfigOptionValue
+          value={this.props.option.value}
+          editing={this.props.option.editing}
+          onSave={this.onSave}
+          onEdit={this.onEdit}
+          onCancelEdit={this.onCancelEdit}
+        />
+        <ConfigOptionTrash
+          optionName={this.props.option.optionName}
+          sectionName={this.props.option.sectionName}
+          onDelete={this.onDelete}/>
+      </tr>
+    );
+  }
+});
+
+var ConfigOptionValue = React.createClass({
+  getInitialState: function () {
+    return {
+      value: this.props.value,
+      editing: this.props.editing,
+      saving: this.props.saving
+    };
+  },
+
+  getDefaultProps: function () {
+    return {
+      value: '',
+      editing: false,
+      saving: false,
+      onSave: () => null,
+      onEdit: () => null,
+      onCancelEdit: () => null
+    };
+  },
+
+  componentWillReceiveProps: function (nextProps) {
+    if (this.props.value !== nextProps.value) {
+      this.setState({ saving: false });
+    }
+  },
+
+  onChange: function (value) {
+    this.setState({ value });
+  },
+
+  onSave: function (value) {
+    if (value !== this.props.value) {
+      this.setState({ saving: true });
+      this.props.onSave(value);
+    } else {
+      this.onCancelEdit();
+    }
+  },
+
+  onEdit: function () {
+    this.props.onEdit();
+  },
+
+  onCancelEdit: function () {
+    this.props.onCancelEdit();
+  },
+
+  getButtons: function () {
+    if (this.state.saving) {
+      return null;
+    } else {
+      return (
+        <span>
+          <button
+            className="btn btn-success fonticon-ok-circled btn-small btn-config-save"
+            onClick={() => this.onSave(this.state.value)}
+          />
+          <button
+            className="btn fonticon-cancel-circled btn-small btn-config-cancel"
+            onClick={this.onCancelEdit}
+          />
+        </span>
+      );
+    }
+  },
+
+  render: function () {
+    if (this.props.editing) {
+      return (
+        <td>
+          <div className="config-value-form">
+            <input
+              onChange={(e) => this.onChange(e.target.value)}
+              defaultValue={this.props.value}
+              disabled={this.state.saving}
+              autoFocus type="text" className="config-value-input"
+            />
+            {this.getButtons()}
+          </div>
+        </td>
+      );
+    } else {
+      return (
+        <td className="config-show-value" onClick={this.onEdit}>
+          {this.props.value}
+        </td>
+      );
+    }
+  }
+});
+
+var ConfigOptionTrash = React.createClass({
+  getInitialState: function () {
+    return {
+      showModal: false
+    };
+  },
+
+  onDelete: function () {
+    this.props.onDelete();
+  },
+
+  render: function () {
+    return (
+      <td className="text-center config-item-trash config-delete-value"
+          onClick={() => this.setState({showModal: true})}>
```

Comment:
I think we need a bit more consistency on how you handle the react events. Sometimes, like here, you do a state change in line. Then other times you call a method, other times you do some inline and then pass a value to a function. I don't mind what you do, but consistency is key